src/Analyzer.Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        public static bool IsEqualsOverride(this IMethodSymbol method)
+        {
+            return method.IsOverride &&
+                   method.Name == WellKnownMemberNames.ObjectEquals &&
+                   method.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                   method.Parameters.Length == 1 &&
+                   method.Parameters[0].Type.SpecialType == SpecialType.System_Object &&
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.GetHashCode.
+        /// </summary>
+        public static bool IsGetHashCodeOverride(this IMethodSymbol method)
+        {
+            return method.IsOverride &&
+                   method.Name == WellKnownMemberNames.ObjectGetHashCode &&  
+                   method.ReturnType.SpecialType == SpecialType.System_Int32 &&
+                   method.Parameters.Length == 0 &&
+                   IsObjectMethodOverride(method);
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides Object.ToString.
+        /// </summary>
+        public static bool IsToStringOverride(this IMethodSymbol method)
+        {
+            return (method != null &&
+                    method.ReturnType.SpecialType == SpecialType.System_String &&
+                    method.Name == WellKnownMemberNames.ObjectToString &&
+                    method.Parameters.Length == 0 &&
+                    IsObjectMethodOverride(method));
+        }
+
+        /// <summary>
+        /// Checks if the given method overrides a method from System.Object
+        /// </summary>
+        private static bool IsObjectMethodOverride(IMethodSymbol method)
+        {
+            var overriddenMethod = method.OverriddenMethod;
+            while (overriddenMethod != null)
+            {
+                if (overriddenMethod.ContainingType.SpecialType == SpecialType.System_Object)
+                {
+                    return true;
+                }
+
+                overriddenMethod = overriddenMethod.OverriddenMethod;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if the given method is a Finalizer implementation.
+        /// </summary>
+        public static bool IsFinalizer(this IMethodSymbol method)
+        {
+            if (method.MethodKind == MethodKind.Destructor)
+            {
+                return true; // for C#
+            }
+
+            if (method.Name != "Finalize" || method.Parameters.Length != 0 || !method.ReturnsVoid)
+            {
+                return false;
+            }
+
+            var overridden = method.OverriddenMethod;
+            if (overridden == null)
+            {
+                return false;
+            }
+
+            for (var o = overridden.OverriddenMethod; o != null; o = o.OverriddenMethod)
+            {
+                overridden = o;
+            }
+
```

Comment:
Shouldn't `IsObjectMethodOverride` also do this loop? (Basically the same question I asked above.) Isn't the purpose of `IsObjectMethodOverride` to make sure that `object.Equals` hasn't been shadowed? In my example above, if `A` declared `public new bool Equals(object obj)`, then you'd want `IsObjectMethodOverride` to return `false`.